src/server/engine_shard_set.cc

```diff
@@ -251,20 +253,36 @@ bool EngineShard::DefragTaskState::CheckRequired() {
     return false;
   }
 
-  const std::size_t threshold_mem = memory_per_shard * GetFlag(FLAGS_mem_defrag_threshold);
-  const double waste_threshold = GetFlag(FLAGS_mem_defrag_waste_threshold);
+  const std::size_t global_threshold = max_memory_limit * GetFlag(FLAGS_mem_defrag_threshold);
+  if (global_threshold > rss_mem_current.load(memory_order_relaxed)) {
+    return false;
+  }
+
+  const auto now = std::chrono::steady_clock::now();
```

Comment:
no need to use `steady_clock` which is relative expensive.
use `time(nullptr)`  which gives you second precision and it's enough